### DISCUSSION TITLE
Get the next hop interface of a peer on a shared instance?

### DISCUSSION THREAD

**Discussion by SebastianObi** [markqvist/Reticulum#397] [Help & Questions] [answered]
Hallo,

I need help with the following requirement:

For the function "remote_identified" at the `Examples/Identify.py` example I need the info about the number of hops and the next hop interface of the peer.
This can be queried with: 
RNS.Transport.hops_to
RNS.Transport.next_hop_interface

The only problem is that my application is connected to a local shared instance, which is already running. Therefore "RNS.Transport.next_hop_interface" returns the LocalInterface, which is not correct at this szenario because the peer is connected to a different interface.

The question now is how to get to the 2nd hop interface or the next hop interface of the shared instance?

Does any reticulum expert here know how I can retrieve this information?

Many thanks.

Just for the background of my "special request" I want to allow request only from a certain number of hops and interface to allow only local clients which are connected to the autointerface to access a resource.

**Comment by faragher** [comment]
Based on the manual's description of the Transport class, my reading of the code, and my understanding of Reticulum, I don't believe this is natively possible.

Reticulum, by design, only knows the next node in the path. The Local Interface is designed to connect to other instances on the same machine, but apparently does act as an additional hop. Since knowing the n+1 hop would require querying the current n hop, I don't think it's possible at this time.

However, if you didn't run the RNS Daemon and simply ran that application as a stand-alone, then there would be no local interface and you could see everything directly. I think.

I expect you are aware of and rejected this already, but have you considered using identity-based trust as opposed to topographically based trust? I'm not certain if you need dynamic access as opposed to curated access, but those systems are already in place, especially if you're using NomadNet, which has both trust and access lists for pages and resources already in place.

It's an interesting concept for access control, but I don't think it can be done without modifying Transport or precluding the use of the daemon/shared instance entirely.

**Comment by markqvist** [comment, marked as answer]
While I don't really recommend basing authentication just on what interface something comes in on, and how many hops, you *can* achieve your goal by using `RNS.Reticulum.get_next_hop_if_name()`. It's not documented yet, but it is a proxy for the equivalent `Transport` method, and works on all parts of a shared instance, both the master and the clients.

Since it's not yet documented, the API for it may change (but probably not, given how simple that method is). Just beware if it suddenly breaks ;)

Also, remember that there may be situations where you cannot count on hop counts, so be *very* careful when doing something like what you described! Again, I don't recommend it, really. But now you know how, anyways ;)

**Reply by SebastianObi** [reply]
Thank you very much,
that was exactly the solution to my requirement.
I am aware that this is of course not secure authentication.
